src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,214 @@
+import { Epsg } from '@basemaps/geo';
+import { fsa } from '@chunkd/fs';
+import { CogTiff } from '@cogeotiff/core';
+import { command, option, positional, string } from 'cmd-ts';
+import { StacCollection, StacItem } from 'stac-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { config, createTiff, registerCli, verbose } from '../common.js';
+import { dataCategories, regions } from './path.constants.js';
+
+export const commandGeneratePath = command({
+  name: 'generate-path',
+  description: 'Generate target path from collection metadata',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+
+    targetBucketName: option({
+      type: string,
+      long: 'target-bucket-name',
+      description: 'Target bucket name, e.g. nz-imagery',
+    }),
+
+    source: positional({
+      type: string,
+      displayName: 'path',
+      description: 'path to source data where collection.json file is located',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    if (args.targetBucketName.startsWith('s3://')) {
+      throw new Error(
+        `target-bucket-name entered as path, did you mean ${args.targetBucketName
+          .replace('s3://', '')
+          .replace('/', '')}?`,
+      );
+    }
+
+    logger.info({ source: args.source }, 'GeneratePath:Start');
+
+    const collection = await fsa.readJson<StacCollection>(fsa.join(args.source, 'collection.json'));
+    if (collection == null) throw new Error(`Failed to get collection.json from ${args.source}.`);
+
+    const tiff = await loadFirstTiff(args.source, collection);
+
+    const category = getCategory(collection);
+    const geographicDescription = getGeographicDescription(collection);
+    const region = getRegion(collection);
+    const event = getEvent(collection);
+    const date = getDate(collection);
+    const gsd = extractGsd(tiff);
+    const epsg = extractEpsg(tiff);
+
+    const target = generatePath(args.targetBucketName, category, geographicDescription, region, event, date, gsd, epsg);
+    logger.info({ duration: performance.now() - startTime, target: target }, 'GeneratePath:Done');
+    return target;
+  },
+});
+
+/**
+ * Generates Target Path based on category
+ *
+ * @param {string} targetBucketName
```

Comment:
generally when you get more than about 3 parameters you should turn it into a object and define a interface

eg  
export interface PathMetadata {
 /** Location of the target bucket */
 targetBucketName: string
}

generatePath(meta: PathMetadata)
```

This makes it much easier to see where strings are transposed in function calls as I have no way of validating that 

```typescript
  generatePath(
        'nz-imagery',
        'satellite-imagery',
        'North Island',
        'new-zealand',
        'Cyclone Gabrielle',
        '2023',
        '0.5m',
        '2193',
      )
 ```
 
 is actually the correct order!